module/VuFind/src/VuFind/Search/Blender/Params.php

```diff
@@ -0,0 +1,655 @@
+            }
+        }
+    }
+
+    /**
+     * Reset the current facet configuration.
+     *
+     * @return void
+     */
+    public function resetFacetConfig()
+    {
+        $this->proxyMethod(__FUNCTION__, func_get_args());
+    }
+
+    /**
+     * Create search backend parameters for advanced features.
+     *
+     * @return ParamBag
+     */
+    public function getBackendParameters(): ParamBag
+    {
+        $result = parent::getBackendParameters();
+        foreach ($this->unsupportedFilters as $backendId => $filters) {
+            if ($filters) {
+                $result->add('fq', "-blender_backend:\"$backendId\"");
+            }
+        }
+        foreach ($this->searchParams as $params) {
+            $backendId = $params->getSearchClassId();
+            if (!is_callable([$params, 'getBackendParameters'])) {
+                throw new \Exception(
+                    "Backend $backendId missing support for getBackendParameters"
+                );
+            }
+
+            // Clone params so that adding any default filters does not affect the
+            // original instance:
+            $params = clone $params;
+            $this->addDefaultFilters($params, $backendId);
+
+            $result->set(
+                "query_$backendId",
+                $params->getQuery()
+            );
+            $result->set(
+                "params_$backendId",
+                $params->getBackendParameters()
+            );
+        }
+        return $result;
+    }
+
+    /**
+     * Add default filters to the given params
+     *
+     * @param BaseParams $params    Params
+     * @param string     $backendId Backend ID
+     *
+     * @return void
+     */
+    protected function addDefaultFilters(BaseParams $params, string $backendId): void
+    {
+        foreach ($this->mappings['Facets']['Fields'] ?? [] as $fieldConfig) {
+            $mappings = $fieldConfig['Mappings'][$backendId] ?? [];
+            $defaultValue = $mappings['DefaultValue'] ?? null;
+            if (null !== $defaultValue) {
+                $translatedField = $mappings['Field'];
+                $filterList = $params->getFilterList();
+                $found = false;
+                foreach ($filterList as $filters) {
+                    foreach ($filters as $filter) {
+                        if ($filter['field'] === $translatedField) {
+                            $found = true;
+                            break;
+                        }
+                    }
+                }
+                if (!$found) {
+                    $params->addFilter("$translatedField:$defaultValue");
+                }
+            }
+        }
+    }
+
+    /**
+     * Proxy a method call to parent class and all backend params classes
+     *
+     * @param string $method Method
+     * @param array  $params Method parameters
+     *
+     * @return mixed
+     */
+    protected function proxyMethod(string $method, array $params)
```

Comment:
While I understand your point, I think I'll ignore the suggestion for now. I feel it would add one level too much for keeping the code easy to understood particularly since there are some differencies to how the methods work.